### PR TITLE
FluentPagination work 

### LINKED ISSUE #1067: Adding component providers to MainLayout causes scrollbars on outer viewport.
Following https://www.fluentui-blazor.net/CodeSetup#add-component-providers using the wasm template, the providers cause scroll bars appear in the overall viewport.

You end up having to add the providers into a nested space such as within the FluentStack within the MainLayout.razor file for the scrollbars to go away.

### LINKED ISSUE #1069: feat: add Pagination Text Template to FluentPaginator
# 🙋 Feature Request

As the Summary is already templated, add the same for the Pagination Text

## 🤔 Expected Behavior

RenderFragment parameter to set the template of Pagination Text (page X of Y)

## 😯 Current Behavior

The pager display text in plain English and in a fixed format

## 💁 Possible Solution

Implement new property as:

`[Parameter] RenderFragment? PaginationTextTemplate {get; set;} `

and change .razor file to take care of presence of this new property

## 🔦 Context

I would like to customize the format and/or language of the pagination text, the same as I can with the summary

## 💻 Examples

N/A

### PR DESCRIPTION
**Pull request by vnbaaij**
- Add PaginationTextTemplate parameter. Fix #1069 
- Update example and add display friendly indices to PaginationState.
- Add clarification to CodeSetup. Fix #1067

### CODE REVIEW

**Review by dvoituron — CHANGES_REQUESTED** [review]

File: src/Core/Components/Pagination/PaginationState.cs [hunk 1]
```diff
@@ -17,6 +17,11 @@ public class PaginationState
     /// </summary>
     public int CurrentPageIndex { get; private set; }
 
+    /// <summary>
+    /// Gets the current 1-based page index. The value cannot be set manually.
+    /// </summary>
+    public int DisplayedCurrentPageIndex => CurrentPageIndex + 1;
```
[the remark this answers is not in the document]
Comment (vnbaaij) [reply src/Core/Components/Pagination/PaginationState.cs:23]: I thought it was perhaps not very clear that the indices (current and last) are 0 based and it makes the template a little cleaner (without the parenthesis and +1). I've made these new ones mbetter discoverable throught the template comment.

[the remark this answers is not in the document]
Comment (dvoituron) [reply src/Core/Components/Pagination/PaginationState.cs:23]: Yes, I understand, but in my opinion, developers shouldn't be surprised that indexes start at zero. Perhaps this can be noted in the comments of the `CurrentPageIndex` property.

[the remark this answers is not in the document]
Comment (vnbaaij) [reply src/Core/Components/Pagination/PaginationState.cs:23]: I'll remove those added `Display...` from the `State`. Comments already mention the indices are zero-based. I'll mention it in the template comments as well:
```
 /// <summary>
 /// Optionally supplies a template for rendering the pagination summary.
 /// The following values can be included:
 /// {your State parameter name}.CurrentPageIndex (zero-basedd, so +1 for the current page number)
 /// {your State parameter name}.LastPageIndex (zero-based, so +1 for the total number of pages)
 /// </summary>
```

**Review by dvoituron — APPROVED** [review]
